src/test/java/uk/gov/companieshouse/acsp/manage/users/integration/UserAcspMembershipIntegrationTest.java

```diff
@@ -720,880 +728,400 @@ void addAcspMemberSuccessWhenInviteeHasRemovedMembershipInDifferentAcsp() throws
+              Arguments.of(witActingOwner, witOwner, "owner", status().isBadRequest()),
+              Arguments.of(witActingOwner, witAdmin, "standard", status().isOk()),
+              Arguments.of(witActingOwner, witAdmin, "admin", status().isOk()),
+              Arguments.of(witActingOwner, witAdmin, "owner", status().isBadRequest()),
+              Arguments.of(witActingOwner, witStandard, "admin", status().isOk()),
+              Arguments.of(witActingOwner, witStandard, "standard", status().isOk()),
+              Arguments.of(witActingOwner, witStandard, "owner", status().isBadRequest()),
+
+              // Admin actions
+              Arguments.of(neiActingAdmin, neiOwner, "standard", status().isBadRequest()),
+              Arguments.of(neiActingAdmin, neiOwner, "admin", status().isBadRequest()),
+              Arguments.of(neiActingAdmin, neiOwner, "owner", status().isBadRequest()),
+              Arguments.of(neiActingAdmin, neiAdmin, "standard", status().isOk()),
+              Arguments.of(neiActingAdmin, neiAdmin, "admin", status().isOk()),
+              Arguments.of(neiActingAdmin, neiAdmin, "owner", status().isBadRequest()),
+              Arguments.of(neiActingAdmin, neiStandard, "admin", status().isOk()),
+              Arguments.of(neiActingAdmin, neiStandard, "standard", status().isOk()),
+              Arguments.of(neiActingAdmin, neiStandard, "owner", status().isBadRequest()),
+
+              // Standard member actions
+              Arguments.of(xmeActingStandard, xmeOwner, "admin", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeOwner, "standard", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeOwner, "owner", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeAdmin, "standard", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeAdmin, "admin", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeAdmin, "owner", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeStandard, "admin", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeStandard, "standard", status().isBadRequest()),
+              Arguments.of(xmeActingStandard, xmeStandard, "owner", status().isBadRequest()));
+    }
 
-    Assertions.assertEquals( UserRoleEnum.STANDARD, updatedAcspMembersDao.getUserRole() );
-    Assertions.assertNotEquals( acspMemberDaos.getFirst().getEtag(), updatedAcspMembersDao.getEtag() );
-  }
+    @ParameterizedTest
+    @MethodSource("membershipUpdateScenarios")
+    void testUpdateAcspMembershipRoleScenarios(
+        AcspMembersDao actor, AcspMembersDao target, String newRole, ResultMatcher expectedStatus)
+        throws Exception {
+      // given
+      acspMembersRepository.insert(List.of(actor, target));
+      final var actingUser = new User();
+      actingUser.setUserId(actor.getUserId());
```

Comment:
And again, replacing the new User() here with:

testDataManger.fetchUserDtos( actor.getUserId() ).getFirst()